src/m_config.c

```diff
@@ -1986,7 +1987,35 @@ static void SetVariable(default_t *def, const char *value)
             break;
 
         case DEFAULT_FLOAT:
-            *def->location.f = (float) atof(value);
+        {
+            // Different locales use different decimal separators.
+            // However, the choice of the current locale isn't always
+            // under our own control. If the atof() function fails to
+            // parse the string representing the floating point number
+            // using the current locale's decimal separator, it will
+            // return 0, resulting in silent sound effects. To
+            // mitigate this, we replace the first non-digit,
+            // non-minus character in the string with the current
+            // locale's decimal separator before passing it to atof().
```

Comment:
This behavior certainly isn't obvious. Would it be easier if instead if `atof()`, you just switch to `sscanf()` and check the result? Then just reject/ignore values that can't be parsed.

Reply:
I think (s)scanf has the same problem with locale. There are extensions which take a locale parameter but nothing standard.

Reply:
The advantage with `sscanf()` will be that it doesn't override variables if it fails to parse the string. However, if we want to preserve user's choice across locale changes, then we'd have to preprocess the string anyway.

Reply:
Use `strtof()` instead?

Reply:
Isn't `atof()` merely a shortcut to `strtof(str,NULL)`?

Reply:
Essentially. The advantage of `strtof` is you can check success/failure by using the second parameter.

Reply:
> I think (s)scanf has the same problem with locale. There are extensions which take a locale parameter but nothing standard.

Oh I understood this too - the config files would continue (sadly) being locale-specific. But at least you could ignore the lines that fail to parse, falling back to the program defaults instead of resetting the values back to zero. Not ideal but probably not so bad.

Reply:
The problem with `sscanf()` is that if it gets passed e.g. `"0,65"` it will read as far as it can parse and then happily assign value `0` to the float variable and return `1` which looks like it successfully parsed the whole string - whereas it gave up on the "decimal comma".

Reply:
I think not fixing the separator would cause some settings to not get saved. Or rather they would be saved and then not loaded correctly but we'd get bug reports saying they're not saved.